app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -50,10 +50,13 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     remove_keys = Array.wrap(extra_keys) + child_keys
 
     record_index = TypedIndex.new(association, find_key)
+    reconnect_index = TypedIndex.new(reconnect_from, find_key)
```

Comment:
Right, forgot to mention it.  This will be heavy, negating the some/all of the win of not saving them :-(
The only other way I could think of was doing one-by-one queries for new records, which is also not good.
It seems refresh fundamentally requires an efficient DB upsert...